x/evm/types/errors.go

```diff
@@ -75,12 +78,38 @@
 
 // NewExecErrorWithReason unpacks the revert return bytes and returns a wrapped error
 // with the return reason.
-func NewExecErrorWithReason(revertReason []byte) error {
-	hexValue := hexutil.Encode(revertReason)
-	reason, errUnpack := abi.UnpackRevert(revertReason)
+func NewExecErrorWithReason(revertReason []byte, vmError string) *revertError {
+	var result []byte
+	result = nil
+	if vmError == vm.ErrExecutionReverted.Error() {
+		result = common.CopyBytes(revertReason)
+	}
+
+	reason, errUnpack := abi.UnpackRevert(result)
+	err := errors.New("execution reverted")
```

Comment:
There are other types of errors, e.g. out of gas?

Reply:
hmm I could not find anything special in geth. Is out of gas including in exection reverted?

Reply:
https://github.com/ethereum/go-ethereum/blob/master/core/vm/errors.go#L31
there are several possible cases for the `VmError`.

Reply:
I see, you are right, I have missread the if condition